python/test_python.py

```diff
@@ -199,4 +200,25 @@ class TestEncodeBase64:
     def test_encode(self, docker_runner):
         docker_runner.run('python encode.py "Hello, world!"')
         docker_runner.container.wait()
-        assert str(docker_runner.container.logs(), 'UTF-8') == 'SGVsbG8sIHdvcmxkIQ==\n'
\ No newline at end of file
+        assert str(docker_runner.container.logs(), 'UTF-8') == 'SGVsbG8sIHdvcmxkIQ==\n'
+
+class TestStreamingStdin:
+    """Test that streaming stdin can be read line by line and can write to stdout
+    without waiting for all lines to arrive
+    Note: this test uses Docker CLI instead of the Python Docker SDK (implemented in the
+    `docker_runner` fixture) since SDK doesn't easily allow for streaming inputs"""
+    def test_stdin(self):
+        # Start a subprocess that runs the script and waits for input
+        script = subprocess.Popen(
+            ['docker', 'run', '-i', 'python-rosetta', 'python', 'streaming_stdin.py'],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            bufsize=1,
```

Comment:
Are any of these arguments unneeded? If so, lets remove them.

Reply:
I don't think we need to set `stderr` unless we want to later read and assert there are no errors in the child process? The default is that the stderr goes to the parent process and that would break our test assertion and we'd see the stderr in the test printout. So dropping that arg.

We **do** want `text` otherwise the default is that everything is in bytes. And we **do** want `bufsize=1` so that the buffer will be flushed at every new line character, which works for the input we're giving in the test.  Then we can get rid of the `script.stdin.flush()` line. The bufsize default is io.DEFAULT_BUFFER_SIZE which seems like it optimizes the buffer based on the os system so it's prob better to be explicit and set it in the Popen constructor like we did.

Reply:
Great! Good work chasing those details down.